src/sql-lexer/src/lexer.rs

```diff
@@ -130,7 +143,42 @@ macro_rules! bail {
 ///
 /// See the module documentation for more information about the lexical
 /// structure of SQL.
+#[cfg(not(target_arch = "wasm32"))]
 pub fn lex(query: &str) -> Result<Vec<(Token, usize)>, LexerError> {
+    lex_inner(query)
+}
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen(js_name = Token, getter_with_clone, inspectable)]
+#[derive(Debug)]
+pub struct JsToken {
+    pub kind: String,
+    pub offset: usize,
+}
```

Comment:
I'd love to avoid forking the return type for `fn lex`. Obviously the `JsError` is unavoidable, but would love for the `Ok` type to be consistent across WASM and Rust. `JsToken` is nicer than the `(Token, usize)` IMO. Can we rename this to `PosToken` or somesuch and make it _the_ return type for Rust, too? I took a quick skim and `lex` is only called in three places or so, so it seems not like a terribly difficult refactor.

Reply:
Done!